subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/SystemValuesAccessor.kt

```diff
@@ -0,0 +1,47 @@
+package com.avito.android.plugin.build_param_check
+
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ProviderFactory
+
+@Suppress("UnstableApiUsage")
+internal class SystemValuesAccessor(private val providers: ProviderFactory) {
```

Comment:
This class has multiple responsibilities:
- Hides reading of system properties. It can be done without grouping them in one god interface.
- Hides formatting and default values. They are not just "system values", but formatted strings with envs.

Maybe renaming to something kinda "BuildEnvironmentInfo" will show an intention better.

Reply:
This is ok as the internal class for hotfix

Reply:
i renamed it